de4a-connector/src/main/java/eu/de4a/connector/as4/client/DomibusGatewayClient.java

```diff
@@ -51,179 +56,180 @@
+            }
+            Document evidence = null;
+            String requestId = "";
+            try {
+                evidence = DOMUtils.byteToDocument(a.getData().bytes());
+                requestId = DOMUtils.getValueFromXpath(DE4AConstants.XPATH_ID, evidence.getDocumentElement());
+            } catch (MessageException e1) {
+                String errorMsg = "Error managing evidence DOM on AS4 response";
+                LOGGER.error(errorMsg, e1);
+                if(a.getContentID() != null && 
+                        a.getContentID().contains(DE4AConstants.TAG_EVIDENCE_RESPONSE)) {
+                    throw (ResponseTransferEvidenceException) ex.withMessageArg(errorMsg);
+                } else {
+                    throw (ResponseErrorException) ex.withMessageArg(errorMsg);
+                }
+            }
+            responsewrapper.setTagDataId(a.getContentID());
+            responsewrapper.setId(requestId);
+            responsewrapper.setResponseDocument(evidence);
+        });
+        return responsewrapper;
+    }
+
+    private List<PartInfo> getAttachments(String idMessageAttached, String idCanonical) {
+        List<PartInfo> attachments = new ArrayList<>();
+        PartInfo partInfo = new PartInfo();
+        PartInfo partInfo2 = new PartInfo();
+        partInfo.setHref(idMessageAttached);
+        partInfo2.setHref(idCanonical);
+
+        Property prop = new Property();
+        prop.setName("MimeType");
+        prop.setValue(MediaType.APPLICATION_XML_VALUE);
+
+        PartProperties props = new PartProperties();
+        props.getProperty().add(prop);
+        partInfo.setPartProperties(props);
+        partInfo2.setPartProperties(props);
+        attachments.add(partInfo);
+        attachments.add(partInfo2);
+
+        return attachments;
+    }
+
+    private MessageOwner buildMessageOwner(LargePayloadType payload, ResponseAndHeader response) {
+        byte[] targetArray = null;
+        try {
+            targetArray = IOUtils.toByteArray(payload.getValue().getDataSource().getInputStream());
+            if (targetArray != null && Base64.isBase64(targetArray)) {
+                targetArray = Base64.decodeBase64(targetArray);
+            }
+
+            Document doc = DOMUtils.byteToDocument(targetArray);
+            MessageOwner messageOwner = new MessageOwner();
+            messageOwner.setMessage(doc.getDocumentElement());
+            String idrequest = null;
+            idrequest = DOMUtils.getValueFromXpath(DE4AConstants.XPATH_ID, doc.getDocumentElement());
+            if (ObjectUtils.isEmpty(idrequest)) {
+                LOGGER.error("EvidenceRequest without requestId");
+            } else {
+                messageOwner.setId(idrequest);
+                SimpleParticipantIdentifier receiver = SimpleIdentifierFactory.INSTANCE.createParticipantIdentifier(
+                        response.getInfo().getUserMessage().getPartyInfo().getTo().getPartyId().getType(), 
```

Comment:
That screams for a NullPointerException ;-)

Reply:
haha, It should be handled by domibus itself. I mean his internal validation wouldn´t process a message without those fields. But, yes, it doesn´t hurt to handle it